Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -223,7 +234,8 @@ public OrderTicket AddOrder(SubmitOrderRequest request)
         public OrderTicket UpdateOrder(UpdateOrderRequest request)
         {
             OrderTicket ticket;
-            if (!_orderTickets.TryGetValue(request.OrderId, out ticket))
+            if (!_openOrderTickets.TryGetValue(request.OrderId, out ticket) &&
+                !_closedOrderTickets.TryGetValue(request.OrderId, out ticket))
```

Comment:
Shouldn't `UpdateOrder` only search open orders?

Reply:
I used both closed and open `Tickets` containers to try to maintain the exact functionally as before: Both `CancelOrder` and `UpdateOrder` are currently returning a specific `OrderResponse` (`InvalidStatus`) when they are sent an order which is already closed. `CancelOrder` is also logging it. In the majority of the cases the `Ticket` should be found in the open container. Nevertheless, we could change this, what do you think?

Reply:
Yeah, agree searching both dictionaries has the same behavior as before. Now that we have separate open/closed dictionaries, perhaps we could give an improved error message like "Order is already filled/cancelled and cannot be updated/cancelled" instead of "Invalid order status".
Another reason for returning immediately if the order is closed is that it would prevent the invalid request from being added to the ticket -- unless this is desirable :)

Reply:
LEAN way is to do this in two steps as you've done. :) Maintain existing behavior; pass tests; merge; update with improved behavior; pass tests; merge.